src/random.cpp

```diff
@@ -33,41 +33,21 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/config.hpp"
 #include "libtorrent/random.hpp"
 
-#include <random>
-
 namespace libtorrent
 {
-	using std::random_device;
-	using std::mt19937;
-	using std::uniform_int_distribution;
-
-#ifdef TORRENT_BUILD_SIMULATOR
-
-	std::uint32_t random(std::uint32_t max)
+	namespace aux
 	{
+#ifdef TORRENT_BUILD_SIMULATOR
 		// make sure random numbers are deterministic. Seed with a fixed number
-		static mt19937 random_engine(0x82daf973);
-		return uniform_int_distribution<std::uint32_t>(0, max)(random_engine);
-	}
-
+		std::mt19937 random_engine(0x82daf973);
 #else
-
-	std::uint32_t random(std::uint32_t max)
-	{
-		// TODO: versions prior to msvc-14 (visual studio 2015) do
-		// not generate thread safe initialization of statics
-		static random_device dev;
-		static mt19937 random_engine(dev());
-		return uniform_int_distribution<std::uint32_t>(0, max)(random_engine);
+		static std::random_device dev;
+		std::mt19937 random_engine(dev());
```

Comment:
Thanks for the very illustrative explanation (actually I was expecting some observation like this). Let me accommodate that.